display_list/display_list_builder.cc

```diff
@@ -653,15 +649,24 @@ void DisplayListBuilder::clipPath(const SkPath& path,
   switch (clip_op) {
     case SkClipOp::kIntersect:
       Push<ClipIntersectPathOp>(0, 1, path, is_aa);
-      if (!current_layer_->clip_bounds.intersect(path.getBounds())) {
-        current_layer_->clip_bounds.setEmpty();
+      if (!path.isInverseFillType()) {
+        intersect(path.getBounds());
       }
       break;
     case SkClipOp::kDifference:
       Push<ClipDifferencePathOp>(0, 1, path, is_aa);
+      if (path.isInverseFillType()) {
```

Comment:
Delete this. kDifference is not an intersection. You can't call intersect here regardless of path type.

Reply:
It might be possible to do a difference computation, but it can't be done on the bounds of the clip being "differenced" and it would involve a lot of round-corner-checking for rrect and path analysis for path. And after all of that you'd have to be really really really sure that it was guaranteed to lop off an entire side of the former clip bounds before you could commit to a change in the conservative estimate of the clip bounds. It's doable for some cases, but I didn't feel it was worth it to work through all of the corner conditions.

Reply:
Hang on, I think I understand what is going on here. You are mapping "kDifference of inverse path" to "kIntersect of the original path".  Hmmm... I think that makes sense now.

Reply:
(I deleted a few comments down below now.)